### ISSUE TITLE
Could not find any assemblies compatible with .NETFramework,Version=v4.5

### ISSUE THREAD

**Issue by lucaritossa** [Redth/Cake.ExtendedNuGet#22]
Today I receive this error because my cake script is downloading the latest version 1.0.0.27 of Cake.ExtendedNuGet.

![image](https://user-images.githubusercontent.com/12578097/30586515-82653b76-9d30-11e7-990a-edb2b4ae70a9.png)

I found that this version seems to be available only for .NET 4.6

![image](https://user-images.githubusercontent.com/12578097/30586450-49021034-9d30-11e7-9499-1745798f33b8.png)

As current workaround I forced my cake script to get the v1.0.0.24 changed this:
`#addin "Cake.ExtendedNuGet"`
to this:
`#addin "nuget:?package=Cake.ExtendedNuGet&version=1.0.0.24"`

**Comment by skovsende** [comment]
1.0.0.27 has been updated to Cake 0.22, which is a .Net 4.6.1 project. Your solution is the correct one until you're ready to update to Cake 0.22 - and when that happens, you should update to Cake.ExtendedNuGet 1.0.0.27.

**Comment by lucaritossa** [comment]
Ok, thank you for your details.

IMHO, the versioning for this project should adopt [semver](http://semver.org/).  
The update to Cake 0.22, with .NET 4.6.1, is a breaking-change and your version should be 2.0.0.

**Comment by skovsende** [comment]
Completely agree, I would suggest making an issue and ping @Redth :)

**Comment by skovsende** [comment]
But do remember that Cake is still not at v1.0, so these kind of breaking changes will happen from time to time.

**Comment by jpreese** [comment]
I am experiencing this as well. Unfortunately the aforementioned fix did not resolve it for me.

```
#addin "nuget:?package=NuGet.Core"
#addin "nuget:?package=Cake.ExtendedNuGet&version=1.0.0.24"
```

**Comment by lucaritossa** [comment]
@jpreese did you clean up the "tools" subfolders where nuget downloaded all stuffs?

@skovsende you are in truth, it seems Cake likes to remain permanently in beta... :( 
A breaking change like this should change major version to Cake too... Do you agree?
I am still convinced that every project must follow semver rules independently

**Comment by skovsende** [comment]
@jpreese Look at #7 - have you tried pinning NuGet.Core?

@lucaritossa Well, I don't think it would make sense to get out of beta before they were using a recent version of Roslyn, which is what was done in 0.22. I'm not deeply into their roadmap, but my gut would tell me we are nearing something that can be called v1.0 - and when that happens, following semver would make sense.

**Comment by rdehuyss** [comment]
Also bumping into something related...

```
PS C:\Projects\CoBen> .\build.ps1 -Target VersionInfo
Preparing to run build script...
Running build script...
Could not load C:\Projects\CoBen\tools\Addins\cake.extendednuget\Cake.ExtendedNuGet\lib\net46\Cake.ExtendedNuGet.dll (missing NuGet.Core, Version=2.14.0.832, Culture=neutral
, PublicKeyToken=31bf3856ad364e35))
Error: Error occurred when compiling build script: C:/Projects/CoBen/build.cake(33,23): error CS0103: The name 'GetPackageReferences' does not exist in the current context
C:/Projects/CoBen/build.cake(299,21): warning CS0168: The variable 'fileNotFound' is declared but never used

I'm running on Cake 0.23. Also tried it with 0.22 but same issue...

**Comment by colinangusmackay** [comment]
@lucaritossa I'm seeing the same issue as @jpreese and I did recently clean out the tools and cake folders (because the bootstrap script wasn't finding cake.exe and that forced it to re-download)

Anyway, I was trying to narrow down the issue so I tried to make a simple cake file to demonstrate the issue, thus:

```
#addin nuget:?package=Cake.MsDeploy

Task("nothing")
  .Does(()=>
  {
    Information("I have succcessfully done nothing useful.");
  });

var target = Argument("target", "nothing");
Information("Running "+target);
RunTarget(target);
```

And that worked! No idea why (it was in the same folder) so using the same cake and tools directory.

But, when I went back to my original script I'm still getting the error message but now moved on to a different add in.

**Comment by markembling** [comment]
I just ran into an issue very similar to this (but with a different addin) and some Googling led me here. The comment from @skovsende was the key - in my case I was still running an older version of Cake (0.19.5) and the add-in I was trying to use wasn't compatible with that. Thought I'd leave this comment in case it helps anyone else.

The key for me was to make sure that Cake was updated - just cleaning up the Cake files as mentioned above by @colinangusmackay wouldn't be enough by itself, as it would more than likely re-download the same old version which was causing the problem before. Instead, the Cake package version in `tools/packages.config` needs to be updated, which will then cause the Cake bootstrap script to re-download the newer version. And on top of that, the addin versions in use need to be compatible with the Cake version. [More info here](https://cakebuild.net/docs/tutorials/pinning-cake-version).

My approach going forward is going to be to pin everything with versions so things can't break later.
